File: src/modules/algorithm/seed/algorithms/binary-tree-traversal/python/exercise.py
from typing import List, Optional

class TreeNode:
    def __init__(self, val: int, left: Optional['TreeNode'] = None, right: Optional['TreeNode'] = None):
        self.val = val
        self.left = left
        self.right = right

def in_order_traversal(root: Optional[TreeNode]) -> List[int]:
    """
    Performs in-order traversal (left, root, right) of a binary tree.
    
    Args:
        root: The root of the binary tree.
        
    Returns:
        A list of node values in in-order.
    """
    # TODO: Implement in-order traversal
    return []

def pre_order_traversal(root: Optional[TreeNode]) -> List[int]:
    """
    Performs pre-order traversal (root, left, right) of a binary tree.
    
    Args:
        root: The root of the binary tree.
        
    Returns:
        A list of node values in pre-order.
    """
    # TODO: Implement pre-order traversal
    return []

def post_order_traversal(root: Optional[TreeNode]) -> List[int]:
    """
    Performs post-order traversal (left, right, root) of a binary tree.
    
    Args:
        root: The root of the binary tree.
        
    Returns:
        A list of node values in post-order.
    """
    # TODO: Implement post-order traversal
    return [] 

File: src/modules/algorithm/seed/algorithms/binary-tree-traversal/javascript/exercise.js
// Define the TreeNode constructor
function TreeNode(val, left = null, right = null) {
  this.val = val;
  this.left = left;
  this.right = right;
}

/**
 * Performs in-order traversal (left, root, right) of a binary tree.
 * @param {TreeNode} root - The root of the binary tree
 * @returns {number[]} Array of node values in in-order.
 */
function inOrderTraversal(root) {
  // TODO: Implement in-order traversal
  return [];
}

/**
 * Performs pre-order traversal (root, left, right) of a binary tree.
 * @param {TreeNode} root - The root of the binary tree
 * @returns {number[]} Array of node values in pre-order.
 */
function preOrderTraversal(root) {
  // TODO: Implement pre-order traversal
  return [];
}

/**
 * Performs post-order traversal (left, right, root) of a binary tree.
 * @param {TreeNode} root - The root of the binary tree
 * @returns {number[]} Array of node values in post-order.
 */
function postOrderTraversal(root) {
  // TODO: Implement post-order traversal
  return [];
}

module.exports = {
  TreeNode,
  inOrderTraversal,
  preOrderTraversal,
  postOrderTraversal,
};
